src/Database/GenericDatabase.php

```diff
@@ -39,6 +42,23 @@ public function fetchFirst(Query $query): ?array
         return $this->fetch($query)[0] ?? null;
     }
 
+    public function withinTransaction(callable $callback): bool
+    {
+        $this->transactionManager->begin();
+
+        try {
+            $callback();
```

Comment:
Should we pass in the transaction manager and/or the database into this closure? Not sure. We can keep it as is for now, but I'd like your input